DOCS-8925 TTL in OSv2 Connector Reference (#1037)
mulesoft/docs-connectors: object-store/1.1/modules/ROOT/pages/object-store-connector-reference.adoc

If not provided, the Mule runtime's default ObjectStoreManager is used. Setting this parameter is only necessary when you want to use a non-default Object Store implementation. For example, if you want to have a store which is backed by Redis or JDBC. |  |
|===

[[objectStorev2]]
=== Object Store (Object Store v2)

NOTE: These settings apply only to xref:object-store::index.adoc[Object Store v2].

[%header%autowidth.spread]
|===
| Field | Type | Description | Default Value | Required

> | Persistent a| Boolean | Whether the store is persistent or transient. | true |

Reviewer: I wonder about that Required column. 
It looks like it is rarely used.
Would it be a good idea to simply replace it with specifying whether a field is required when naming the field?
Or, perhaps we should use "Yes" and "No," so that a blank cell does not appear? I always wonder if audiences question a blank cell, which is why most depts. I have worked in use "n.a." to reassure readers that something isn't just missing....

| Max Entries a| | *Not configurable for Object Store v2* 

The maximum number of entries is unlimited.|  |
| Entry Ttl a| Number a| The entry timeout.

Ignored if the value of `expirationInterval` is less than or equal to `0`.

* Minimum value: 1 second
* Maximum value: 30 days

If the field contains: 

* No value
+
** Mule versions 4.2.1 and later: Rolling TTL
** Mule versions earlier than 4.2.1: Static TTL of 30 days by default

* Less than or equal to `0`
+

> TTL set to the maximum value (2592000 seconds/30 days).

Reviewer 2: Also, specify that this will not be rolling ttl in this case.

Author: added static

* Positive number
+
TTL set to the specified value (between 1 and 2592000 seconds)

*Include link to TTL in FAQ and/or TTL task*
| 2592000 seconds (30 days)  |

> | Entry Ttl Unit a| Enumeration, one of:

Reviewer: Ttl lettercasing OK here?

Author: It's a mishmash of the field/parameter names. I'll use field names, which are also questionably capitalized.

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS

> ** DAYS a| The unit for the entry's time to live (`entryTtl`) | SECONDS |

Reviewer: Should this end with a period; is it a peer with the other fragments that end in periods above?

| Expiration Interval a| Positive number a| How frequently the expiration thread should run.

Set `expirationInterval` to a positive value that less than the value of `entryTtl`.

If `expirationInterval` is set to a value less than or equal to zero, `entryTtl` is ignored. 

The `expirationInterval` field requires that the `entryTtl` field is set;
otherwise, it is ignored.
| 1 minute |
| Expiration Interval Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS a| The expiration interval (`expirationInterval`). | MINUTES |

> | Config Ref a| <<ObjectStoreConnector>> a| A reference to an `os:config` element used to declare this Object Store.

Reviewer: GLOBALLY be sure that we are making the distinction between object store and Object Store clear and consistent.

> If not provided, the Mule runtime's default ObjectStoreManager is used. Setting this parameter is only necessary when you want to use a non-default Object Store implementation. For example, if you want to have a store which is backed by Redis or JDBC. |  |

Reviewer: ```suggestion
If not provided, the Mule runtime's default ObjectStoreManager is used. Setting this parameter is necessary only when you want to use a nondefault Object Store implementation: for example, if you want to have a store that is backed by Redis or JDBC. |  |
```

If not provided, the Mule runtime's default ObjectStoreManager is used. Setting this parameter is only necessary when you want to use a non-default Object Store implementation. For example, if you want to have a store which is backed by Redis or JDBC. |  |
|===

[[objectStorev2]]
=== Object Store (Object Store v2)

NOTE: These settings apply only to xref:object-store::index.adoc[Object Store v2].

[%header%autowidth.spread]
|===
| Field | Type | Description | Default Value | Required
| Persistent a| Boolean | Whether the store is persistent or transient. | true |
| Max Entries a| | *Not configurable for Object Store v2* 

The maximum number of entries is unlimited.|  |
| Entry Ttl a| Number a| The entry timeout.

Ignored if the value of `expirationInterval` is less than or equal to `0`.

* Minimum value: 1 second
* Maximum value: 30 days

If the field contains: 

* No value
+
** Mule version 4.2.x, 4.3.0, and later: Rolling TTL
** Mule versions earlier than 4.2.1: Static TTL of 30 days by default

* Less than or equal to `0`
+
TTL set to the maximum value (2592000 seconds/30 days).
* Positive number
+

> TTL set to the specified value

Reviewer 3: maybe add something  like "but not more than max TTL (2592000 seconds/30 days)"

Author: added (between 1 and 2592000 seconds)

> // ** Mule version 4.2.x, 4.3.0, and later: TTL set to the specified value and creates a rolling TTL. 

Reviewer 3: I realized that 4.2.x is not correct. Because there is 4.2.0 - that we never touched.

So should be >=4.2.1, 4.3.x. Not sure what is the best to write that

Author: 4.2.1 and later
